### ISSUE TITLE
Additional view fields (for large grid views)

### ISSUE THREAD

**Issue by constantinpape** [mobie/mobie.github.io#62]
I think we should add a flag `visible ` to `imageDisplay` and `segmentationDisplay`, that determines whether the corresponding layer in the viewer will be shown or not when the view is loaded. This should be optional and default to true.
I think we discussed this already and decided against this to keep complexity low; however for large grid views it would be important to have this: there the best way to show a full plate is to add all sources to a single view. But, in order to not load too many images upon loading the view, it would be very helpful if only a single of the layers could be activated.

(For the same use-case, it would also be good to do something similar with the segmentation tables, but I want to check how it works right now and can't due to https://github.com/mobie/mobie.github.io/issues/58#issuecomment-891041547)

cc @tischi @K-Meech

**Comment by tischi** [comment]
I think adding this field makes sense, because it is something that exists in the UI state and thus would be good to reflect in the JSON. If you add it as an example somewhere I can implement it on the Java side.

**Comment by constantinpape** [comment]
> If you add it as an example somewhere I can implement it on the Java side.

Ok, I would wait till you fix https://github.com/mobie/mobie.github.io/issues/58#issuecomment-891041547 and then add it to that grid view.

**Comment by constantinpape** [comment]
@tischi I have added two fields now `visible` (determines if sources are active in viewer) in `imageDisplay` and `segmentationDisplay` and `showTable` (determines if table GUI is active) in `segmentationDisplay` and `annotationDisplay`.

I have updated the grid view in the covid if project accordingly.

**Comment by constantinpape** [comment]
@tischi For completeness I think we should also add the `visible` field to the `annotationDisplay`, to control the visibility of the colored overlay for these. What do you think?

**Comment by tischi** [comment]
> For completeness I think we should also add the visible field to the annotationDisplay, to control the visibility of the colored overlay for these. What do you think?

yes, in fact I sort of had it there already because I added the `visible` field to `SourceDisplay` which is the base class of all the display classes. I just forgot to use it somewhere else in the code where the display is shown. But next time I update MoBIE-beta it will be there.
